lib/gollum/helpers.rb

```diff
@@ -23,7 +27,8 @@ def extract_name(file_path)
 
       # File.basename is too eager to please and will return the last
       # component of the path even if it ends with a directory separator.
-      ::File.basename(file_path)
+      ext = ::File.extname(file_path)
+      return ::File.basename(file_path, ext), ext
```

Comment:
I had to first chop the file extension of (via `extract_name`) and then join it back to the name (via `join_page_name`) because currently, if we pass e.g. `name.md` to `gollum-lib`, the page in the repo ends up being called `name.md.md`. This makes the code in `app.rb` even uglier, but I propose we first merge it like this before we can do a more extensive rewrite (that will probably have to include the method signatures of `write_page` and such in `gollum-lib`).